libsolidity/codegen/LValue.cpp

```diff
@@ -357,38 +357,45 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 				"Struct assignment with conversion."
 			);
 			solAssert(!structType.containsNestedMapping(), "");
-			solAssert(sourceType.location() != DataLocation::CallData, "Structs in calldata not supported.");
-			for (auto const& member: structType.members(nullptr))
+			if (sourceType.location() == DataLocation::CallData)
 			{
-				// assign each member that can live outside of storage
-				TypePointer const& memberType = member.type;
-				solAssert(memberType->nameable(), "");
-				TypePointer sourceMemberType = sourceType.memberType(member.name);
-				if (sourceType.location() == DataLocation::Storage)
+				m_context << Instruction::DUP2 << Instruction::DUP2;
+				m_context.callYulFunction(m_context.utilFunctions().updateStorageValueFunction(structType, &sourceType, 0), 2, 0);
+			}
+			else
+			{
+				for (auto const& member: structType.members(nullptr))
 				{
-					// stack layout: source_ref target_ref
-					pair<u256, unsigned> const& offsets = sourceType.storageOffsetsOfMember(member.name);
-					m_context << offsets.first << Instruction::DUP3 << Instruction::ADD;
+					// assign each member that can live outside of storage
```

Comment:
This is not relevant anymore, all of them have to.

Reply:
So maybe we don't need the loop anymore at all?

Reply:
I did call yul function only for copying from calldata to storage. I think we should leave the rest of function as is, and maybe later when we finalize all of the conversions in sol2Yul call for others. So, that is why I prefer not to change old code here. Also, diff looks really weird here, as I only added part related to calldata.

I think we should leave the loop as it is related to something else, not the purpose of the calldata to storage part...

Reply:
ok!